spec/rspec/core/example_spec.rb

```diff
@@ -732,4 +732,19 @@ def expect_pending_result(example)
       expect(ex).to pass
     end
   end
+
+  describe "exposing the examples reporter" do
+    it "returns a null reporter when the example hasnt run yet" do
+      example = RSpec.describe.example
+      expect(example.reporter).to be_a RSpec::Core::NullReporter
+    end
+
+    it "returns the reporter used to run the example when executed" do
+      reporter = double(:reporter).as_null_object
+      group = RSpec.describe
+      example = group.example
+      example.run group.new, reporter
+      expect(example.reporter).to eq reporter
```

Comment:
I think `be` is more appropriate here -- we want it to be the exact same reporter object, not simply an object that is equal according to `==` semantics.